Panic with invalid input

how to reproduce

```go
package expr_test

import (
	"testing"

	"github.com/antonmedv/expr"
)

func TestPanic(t *testing.T) {
	const input = "4..0"
	_, err := expr.Compile(input)
	t.Log(err)
}
```

```
$ go test . -run TestPanic -v
=== RUN   TestPanic
--- FAIL: TestPanic (0.00s)
panic: runtime error: makeslice: len out of range [recovered]
        panic: runtime error: makeslice: len out of range

goroutine 6 [running]:
testing.tRunner.func1.1(0x6bf8e0, 0x766dc0)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.15.7_1/libexec/src/testing/testing.go:1072 +0x30d
testing.tRunner.func1(0xc000001b00)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.15.7_1/libexec/src/testing/testing.go:1075 +0x41a
panic(0x6bf8e0, 0x766dc0)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.15.7_1/libexec/src/runtime/panic.go:969 +0x1b9
github.com/antonmedv/expr/optimizer.(*constRange).Exit(0x1326408, 0xc00000e480)
        /home/heijo/ghq/github.com/antonmedv/expr/optimizer/const_range.go:22 +0xe5
github.com/antonmedv/expr/ast.(*walker).walk(0xc000073e78, 0xc00000e480)
        /home/heijo/ghq/github.com/antonmedv/expr/ast/visitor.go:45 +0x4be
github.com/antonmedv/expr/ast.Walk(...)
        /home/heijo/ghq/github.com/antonmedv/expr/ast/visitor.go:18
github.com/antonmedv/expr/optimizer.Optimize(0xc00000e480, 0xc000032180, 0x7724e0, 0x6a6620)
        /home/heijo/ghq/github.com/antonmedv/expr/optimizer/optimizer.go:35 +0x268
github.com/antonmedv/expr.Compile(0x70175f, 0x4, 0x0, 0x0, 0x0, 0xc000062770, 0x48ff46, 0x602a76d2)
        /home/heijo/ghq/github.com/antonmedv/expr/expr.go:167 +0x291
github.com/antonmedv/expr_test.TestPanic(0xc000001b00)
        /home/heijo/ghq/github.com/antonmedv/expr/panic_test.go:11 +0x4b
testing.tRunner(0xc000001b00, 0x72ea38)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.15.7_1/libexec/src/testing/testing.go:1123 +0xef
created by testing.(*T).Run
        /home/linuxbrew/.linuxbrew/Cellar/go/1.15.7_1/libexec/src/testing/testing.go:1168 +0x2b3
FAIL    github.com/antonmedv/expr       0.004s
FAIL

```

--- a/expr_test.go
+++ b/expr_test.go
@@ -813,6 +813,14 @@ func TestExpr(t *testing.T) {
 			6,
 		},
 		{
+			`4 in 5..1`,
+			false,
+		},
+		{
+			`4..0`,
+			[]int{},
+		},
+		{
 			`MapArg({foo: "bar"})`,
 			"bar",
 		},

--- a/optimizer/const_range.go
+++ b/optimizer/const_range.go
@@ -14,6 +14,14 @@ func (*constRange) Exit(node *Node) {
 			if min, ok := n.Left.(*IntegerNode); ok {
 				if max, ok := n.Right.(*IntegerNode); ok {
 					size := max.Value - min.Value + 1
+					// In case the max < min, patch empty slice
+					// as max must be greater than equal to min.
+					if size < 1 {
+						Patch(node, &ConstantNode{
+							Value: make([]int, 0),
+						})
+						return
+					}
 					// In this case array is too big. Skip generation,
 					// and wait for memory budget detection on runtime.
 					if size > 1e6 {

--- a/vm/runtime.go
+++ b/vm/runtime.go
@@ -220,6 +220,9 @@ func exponent(a, b interface{}) float64 {
 
 func makeRange(min, max int) []int {
 	size := max - min + 1
+	if size <= 0 {
+		return []int{}
+	}
 	rng := make([]int, size)
 	for i := range rng {
 		rng[i] = min + i

